src/vm/ceeload.h

```diff
@@ -3686,6 +3686,7 @@ class ReflectionModule : public Module
     // If true, then only other transient modules can depend on this module.
     bool m_fIsTransient;
 
+#if !defined DACCESS_COMPILE && !defined CROSSGEN_COMPILE
```

Comment:
The sanitizer is doing de-optimizations that are introducing references to unreachable code - that is not there in some build configs (dac, crossgen).